Initial working version

keitwb opened the pull request:
See README for details.

Any feedback on cleaning this up and making it more maintainable is welcome.  This was kind of hacked together in a two day span but let me know any issues you see.

Commit:
Initial working version

mjtieman commented on collect-libs.sh:
```diff
@@ -0,0 +1,61 @@
+# LD_LIBRARY_PATH gets priority over default system paths
+export LD_LIBRARY_PATH=$INSTALL_DIR/lib:$INSTALL_DIR/lib/x86_64-linux-gnu
+new_deps=$(find_deps)
+export LD_LIBRARY_PATH=
```
Why the empty assignment?

keitwb replied:
To restore it back to the system default so that it doesn't break shell commands by trying to link system binaries that the script uses (e.g. `mkdir`, stuff that all distros have) against our bundled libs (esp. `libc` tends to break stuff more than anything).

jrcamp replied:
You could just do:

`new_deps=$(LD_LIBRARY_PATH=$INSTALL_DIR/lib:$INSTALL_DIR/lib/x86_64-linux-gnu find_deps)`

instead of setting/unsetting.

mjtieman commented on Dockerfile:
```diff
@@ -0,0 +1,161 @@
+    cp -r /usr/lib/python2.7 /opt/collectd/lib/python2.7
+
+# jq and curl are used to get the AWS unique id out from the AWS metadata endpoint
+# The curl binary needs a different libcurl so so copy it in too
```
one "so"

Commit:
adding quotes around hostname

Commit:
Use Gomplate instead of sed for config templates

- Using netcat to get AWS id instead of curl since libcurl was giving warnings
  on centos machines
- Allowing system metrics to be disabled through envvar

mjtieman commented on README.md:
```diff
@@ -0,0 +1,104 @@
+Currently this doesn't support Java plugins.  This should be possible to add
+at some point.
+
+## Installation and Running
```
Might want to list the optional environment vars HOSTNAME, NO_SYSTEM_METRICS and extra dimensions.

Commit:
Adding config options to README

Commit:
Updating docs and adding versioning scheme

jrcamp commented on templates/collectd.conf:
```diff
@@ -0,0 +1,79 @@
+{{ if (getenv "HOSTNAME") }}
+Hostname   "{{ .Env.HOSTNAME }}"
+{{ else }}
+FQDNLookup   true"
```
Doesn't look like this would parse (shouldn't have " at the end).

jrcamp commented on run.sh:
```diff
@@ -0,0 +1,82 @@
+done
+
+SFX_INGEST_URL=${SFX_INGEST_URL:-https://ingest.signalfx.com}
+[[ -z $API_TOKEN ]] && echo 'API_TOKEN envvar must be given!' >&2 && exit 1
```
docs wants to standardize on ACCESS_TOKEN

jrcamp commented on collect-libs.sh:
```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+set -ex
```
I'd suggest doing -o pipefail as well.

Commit:
Changing name of API_TOKEN to ACCESS_TOKEN

 - Minor tweaks
